examples/armv4t/emu.rs

```diff
@@ -25,10 +26,12 @@ pub struct Emu {
 
     pub(crate) watchpoints: Vec<u32>,
     pub(crate) breakpoints: Vec<u32>,
+    pub(crate) files: [Option<std::fs::File>; FD_MAX as usize],
```

Comment:
(ignore the FdKind stuff - it won't be relevant once you switch to using `HostIoHelper`)

Reply:
I consider about `HashMap` at first, but if there is a situation that fds `1,2,3` are opened, and then `2` is closed, how could we fill `2` with new file if we use `HashMap`?

Reply:
Ah, fair point.

Two schools of thought:
1. have a second "free list" which is a Vec<u32> that you push any freed ids onto, which you can then check before assigning new ids. This trades memory for constant-time id assignment.
2. use a Vec<Option<u32>> with a linear scan when assigning fds. 

Personally, I would lean towards option 1, since this code will be running on `std` platforms with a bunch of memory, where I'm willing to trade memory for better perf.

Reply:
Am I using the second method now?
As for the first, I think do this thing with an `HashMap` and an `Vec` is really not elegant.

Reply:
Since we've punted the implementation of a `HostIoHelper` to another PR, this doesn't matter too much. We can argue the merits / cons of this approach in another PR.

That said, one thing I _would_ like you to do is swap out the current fixed-size array implementation for a `Vec<Option<u32>>`. I don't see any reason why we need a MAX_FD constant - it seems unnecessarily restrictive.